### ISSUE TITLE
Changing settings in Dark Mode in Preferences should not reset Style Configurator

### ISSUE THREAD

**Issue by Edditoria** [notepad-plus-plus/notepad-plus-plus#10196]
### Description of the Issue

In v8.1.2, changing settings in Dark Mode in Preferences will reset Style Configurator to "DarkModeDefault". (Should it be "Dark Mode Default?)
They should not interact in this way.

### Steps to Reproduce the Issue

1. Open Preferences via Settings.
2. Click "Dark Mode".
3. Click something including:
    - "Enable dark mode"
    - Select any tone
    - Disable "Enable dark mode"
4. Change "Style Configurator" again.
5. Loop step 3 and 4.

### Expected Behavior

User can select their favourite tone (app-theme) and styler (syntax-theme) separately.
These two type of configs are in different dialog, anyway.

### Actual Behavior

When changing settings related to dark mode, the styler switches to "DarkModeDefault".
When disable "Enable dark mode", the styler switches to "Default (styler.xml)".

**Comment by ozone10** [comment]
related (maybe duplicate) to https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10049

**Comment by donho** [comment]
@Edditoria 
I'm looking for a way to switch to `markdown.zenburn.udl.xml` :
https://github.com/Edditoria/markdown-plus-plus/blob/master/udl/markdown.zenburn.udl.xml

if dark mode is enabled.

**Comment by Edditoria** [comment]
Let me evalue how dark mode should work:

- There are 3 windows related to color appearance:
  1. Preferences - Dark Mode.
  2. Style Configurator.
  3. UDL.
- In contrast of syntax themes, users may often switch between dark mode and light mode (when dark mode is disable).
- You may want to create tones for light mode in future.
- Users have their favourite themes in light and dark, and the themes are **most possibly from different authors**.
- So, it is not good to integrate different UDLs in one file.
- User may want to use a dark theme in light mode.
- Pre-defined dark/light in UDL does not work either.
- Manually change configs in UDL may be too complicated for users, especially when they update their UDLs in bulk.

To make it simple and easy to understand for user:

- There will be 2 main categories: dark mode and light mode.
- There will be user-selectable syntax themes: 1 for dark (apply to all tones), and 1 for light.
- Syntax theme:
  - ~~In Preferences - Dark Mode, users can select 1 syntax theme for dark and 1 for light.~~ (Sorry it is Microsoft-way)
  - For NPP, there are 2 setting variables: user_selected_theme(_light?) and user_selected_theme_dark
  - When dark mode is triggered, NPP will automatically update the user-selected theme in Style Configurator (Assume that user had opened the window).
  - **Happy ending**: Users don't need to switch multiple setting windows. NPP will do it for users.
- UDLs:
  - ~~Simple. 2 folders: `dark` and `light`.~~ (Sorry it is Microsoft-way)
  - Create 1 more folder when app is updated/installed: `<%npp%\userDefineLangs\dark mode\>`.
  - If NPP match a UDL in dark folder, use that.
  - Otherwise, load the UDL file in `<userDefineLangs\*>` (fallback, default, whatever).
  - NPP should update the opened documents when dark mode or color tone is changed.
  - **Happy ending**: Users move or copy the existing UDL files to the dark mode folder. Simple management.

@donho Put a dark UDL and name it `markdown._preinstalled.udl.xml`.
That `_` is important: Let NPP loading other UDLs in prior of preinstalled file.
Also, users don't need to remove the file after every app update. Less annoying!
I can create one for you, if you want.

I just type what pop up in my head. So, anyone can challenge me :smile:

**Comment by donho** [comment]
@Edditoria 
The light mode will be the system default one. There won't be the different tons for the light mode.
Thanks to the fix done by @ozone10 , the style doesn't change while the dark mode is not switched OFF, so one of the most important problems pointed out in this issue has been solved.

It remains an issue that there's no markdown.zenburn.udl.xml in the package and no way to apply this UDL file on the fly while user switches Notepad++ into dark mode.

In this PR, the above remained issue should be solved. I didn't use the idea of dark mode directory - it will add a lot of modification and implementation in Notepad++. Instead, a new attribute `darkModeTheme="yes"` is added manually in `UserLang` tag for distinguishing from the light mode UDL file. 

I did some tests, it works fine. However, though the dark mode is detected dynamically, and all new opened MD files will be applied to the right UDL, the opened MD files (opened before switching mode) remain the previous UDL. User has to close & reopen them to get the correct UDL applied.

This fix will be in v8.1.3.
